src/treemodel/selection.js

```diff
@@ -131,12 +141,44 @@ export default class Selection {
 	 * @returns {Array.<LiveRange>}
 	 */
 	getRanges() {
-		return this._ranges.slice();
+		return this._ranges.length ? this._ranges.slice() : [ this._getDefaultRange() ];
+	}
+
+	/**
+	 * Returns the first range in the selection. First range is the one which {@link core.treeModel.Range#start start} position
+	 * {@link core.treeModel.Position#isBefore is before} start position of all other ranges (not to confuse with the first range
+	 * added to the selection).
+	 *
+	 * @returns {core.treeModel.Range}
+	 */
+	getFirstRange() {
+		let first = null;
+
+		for ( let i = 0; i < this._ranges.length; i++ ) {
+			let range = this._ranges[ i ];
+
+			if ( !first || range.start.isBefore( first.start ) ) {
+				first = range;
+			}
+		}
+
+		return first && Range.createFromRange( first ) || this._getDefaultRange();
```

Comment:
Please use `?:` :P. If anytime in the future `Range.createFromRange()` would return a falsy value there would be a very nasty bug here. I've really seen those.